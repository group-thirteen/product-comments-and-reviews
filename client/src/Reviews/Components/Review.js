/* eslint-disable no-unused-vars */

import React from 'react'
import styles from './components.css'
import ReviewTitle from './ReviewTitle'
import ReviewBody from './ReviewBody'
import ReviewUsername from './ReviewUsername'
import ReviewRating from './ReviewRating'
import ReviewDate from './ReviewDate'

const Review = (props) => {
  return (
    <div className={styles.review}>
      <ReviewRating rating={props.review.score} />
      <ReviewTitle title={props.review.title} />
      <ReviewBody body={props.review.body} />
      <ReviewUsername username={props.review.username} />
      <ReviewDate date={props.review.date} />
    </div>
  )
}

export default Review
