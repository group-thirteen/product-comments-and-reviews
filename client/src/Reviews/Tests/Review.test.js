/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
import React from 'react'
import { shallow, mount } from 'enzyme'
import Review from '../Components/Review.js'
import { createSerializer } from 'enzyme-to-json'
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }))
const testReview = {
  username: 'Elias',
  title: 'Title',
  body: 'Body',
  score: '5'
}

describe('Review', () => {
  const wrapped = mount(<Review review={testReview} />)
  it('should render the review title', () => {
    expect(wrapped).toMatchSnapshot()
  })
})
