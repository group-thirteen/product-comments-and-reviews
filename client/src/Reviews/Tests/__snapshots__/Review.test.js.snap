// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should render the review title 1`] = `
<div
  className="review"
>
  <div
    className="reviewRating"
  >
    5
     / 5
  </div>
  <div
    className="reviewTitle"
  >
    Title
  </div>
  <div
    className="reviewBody"
  >
    Body
  </div>
  <div
    className="reviewUsername"
  >
    Elias
  </div>
  <div
    className="reviewDate"
  >
    Mar 22, 2020
  </div>
</div>
`;
